doc/decisions/frontend_e2e_testing.md

```diff
@@ -0,0 +1,121 @@
+# E2E Testing
+
+## Problem
+
+There is currently no existing e2e tests locally and in the pipeline.
+This allows for increased frontend bugs and regression.
+It lacks profound decision making which framework to pick.
+
+## Constraints
+
+1. [Behavior-driven development](https://www.selenium.dev/documentation/test_practices/testing_types/#behavior-driven-development-bdd)
+   - a) We want to have [.feature](https://toolsqa.com/cucumber/cucumber-jvm-feature-file/) files and stick to the [Gherkin syntax](https://cucumber.io/docs/gherkin/reference/).
+2. [Page Object Design Pattern](https://www.selenium.dev/documentation/test_practices/encouraged/page_object_models/)
+   - a) A programming language that supports classes
+3. The library must be free software.
+4. A minimalistic and simplistic solution (little lines/characters).
+5. The framework must have an active community.
+6. Headless browser support so we can run the tests in our automated pipeline.
+
+## Assumptions
+
+1. Understanding of BDD practices and principles, including writing feature files using the Gherkin syntax.
+2. Familiarity with the Page Object Design Pattern and its implementation for creating modular and maintainable test code.
+3. Knowledge of end-to-end testing concepts and best practices.
+4. Basic understanding of Python programming language and its syntax.
+5. Having heard of pytest and its usage for test execution and assertions.
+
+## Solutions
+
+### Alternative [Java](https://www.java.com/de/) + [Selenium](https://www.selenium.dev)
+
+Java with Selenium is a popular and well-established choice for web test automation.
+While it remains a reliable option, there are personal preferences and concerns about Oracle's licensing changes, and there are other languages that can do the same with less boilerplate.
```

Comment:
Can you point me to something about this license change? It looks to me it uses Apache 2.0?